src/library_glfw.js

```diff
@@ -641,6 +676,31 @@ var LibraryGLFW = {
       }
     },
 
+    setJoystickCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.joystickFunc = cbfun;
+    },
+
+    refreshJoystick: function(joy) {
+      // TODO: refresh all joysticks, and call from render loop?
+      var j = GLFW.active.joys[joy];
+      if (!j) return;
+
+      var gamepad = navigator.getGamepads()[j.index];
```

Comment:
See https://github.com/kripken/emscripten/pull/4292 for a similar change that was done for the HTML5 API counterpart, and also https://github.com/kripken/emscripten/pull/4921 which might be of interest.